tools/src/main/java/org/apache/kafka/tools/TopicCommand.java

```diff
@@ -537,6 +544,18 @@ public Map<TopicPartition, PartitionReassignment> listAllReassignments(Set<Topic
         }
 
         public void describeTopic(TopicCommandOptions opts) throws ExecutionException, InterruptedException {
+            try {
+                describeTopic(opts, true);
+            } catch (Exception e) {
+                if (e.getMessage().contains("UnsupportedVersionException")) {
```

Comment:
This seems a little fragile. Is there a reason we can't check the actual class type of the exception (or its children, if necessary)?

Reply:
Refactored.